### ISSUE TITLE
Overriding empty value of ComboBox does not work if count of items > pagesize

### ISSUE THREAD

**Issue by TatuLund** [vaadin/flow-components#2520]
Vaadin 14.8.1

Code to reproduce at the end the ticket

Sequence to reproduce

Type in filter text field, say "test 5" and tab out

You will see "test 5" selected and marked with check mark Notification shows the value.

Click text field again and empty it using delete key and again tab out

You will see from Notification that value is correctly null as empty string is set to represent null

Click drop down open and you will see that "test 5" is still checked

Expected behavior would be that "test 5" is not checked. If you modify the example to have e.g. only 10 items, ComboBox works correctly, alternatively you can set the pagesize to be say 100, and it will again work as pagesize is larger than count of items.

```
@Route("combo-box-null-bug")
public class ComboBoxNullBug extends VerticalLayout {

    public ComboBoxNullBug() {
        MyCombo combo = new MyCombo();
        // Uncomment for workaround
        // combo.setPageSize(100);

        List<String> items = new LinkedList<>();
        IntStream.range(1, 60).forEach(value -> {
            items.add("test " + value);
        });
        combo.setItems(items);

        combo.addValueChangeListener(e -> {
            Notification.show(String.format("event: %s, combo: %s",
                    e.getValue(), combo.getValue()));
        });

        add(combo);
    }

    class MyCombo extends ComboBox<String> {
        @Override
        public String getEmptyValue() {
            return "";
        }
    }
}
```

**Comment by TatuLund** [comment]
Could this be fixed by https://github.com/vaadin/flow-components/pull/2529 ?

**Comment by web-padawan** [comment]
> Click drop down open and you will see that "test 5" is still checked

Yes this was fixed by #2529. Is there anything else in this issue that needs to be addressed?

**Comment by TatuLund** [comment]
> Yes this was fixed by #2529. Is there anything else in this issue that needs to be addressed?

@web-padawan I think not.

**Comment by web-padawan** [comment]
Thanks, closing as fixed.
